core/blockchain.go

```diff
@@ -3460,29 +3474,36 @@ func (bc *BlockChain) PCRC(header *types.Header, headerOrder int) (types.PCRCTer
 		// PTZ and RTZ are essentially a signaling mechanism to know that we are building on the right terminal header.
 		// So running this only on a coincident block makes sure that the zones can move and sync past the coincident.
 		// Just run RTZ to make sure that its linked. This check decouples this signaling and linking paradigm.
-
-		PTZ, err := bc.PreviousCanonicalCoincidentOnPath(header, slice, params.PRIME, params.ZONE, true)
-		if err != nil {
-			return types.PCRCTermini{}, err
+		if headerOrder < params.REGION {
+			fmt.Println("PCRC Running PTZ")
+			PTZ, err := bc.PreviousValidCoincidentOnPath(header, slice, params.PRIME, params.ZONE, true)
+			fmt.Println("Hash: PTZ", PTZ.Hash(), "error:", err)
+			if err != nil {
+				return types.PCRCTermini{}, err
+			}
+			PCRCTermini.PTZ = PTZ.Hash()
 		}
-		PCRCTermini.PTZ = PTZ.Hash()
 
-		RTZ, err := bc.PreviousCanonicalCoincidentOnPath(header, slice, params.REGION, params.ZONE, true)
-		if err != nil {
-			return types.PCRCTermini{}, err
+		if headerOrder < params.ZONE {
+			fmt.Println("PCRC Running RTZ")
+			RTZ, err := bc.PreviousValidCoincidentOnPath(header, slice, params.REGION, params.ZONE, true)
+			fmt.Println("Hash: RTZ", RTZ.Hash(), "error:", err)
+			if err != nil {
+				return types.PCRCTermini{}, err
+			}
+			PCRCTermini.RTZ = RTZ.Hash()
 		}
-		PCRCTermini.RTZ = RTZ.Hash()
 
 		return PCRCTermini, nil
 	}
 	return types.PCRCTermini{}, errors.New("running in unsupported context")
 }
 
-// PreviousCanonicalCoincidentOnPath searches the path for a cononical block of specified order in the specified slice
+// PreviousValidCoincidentOnPath searches the path for a cononical block of specified order in the specified slice
 //     *slice - The zone location which defines the slice in which we are validating
 //     *order - The order of the conincidence that is desired
 //     *path - Search among ancestors of this path in the specified slice
-func (bc *BlockChain) PreviousCanonicalCoincidentOnPath(header *types.Header, slice []byte, order, path int, fullSliceEqual bool) (*types.Header, error) {
+func (bc *BlockChain) PreviousValidCoincidentOnPath(header *types.Header, slice []byte, order, path int, fullSliceEqual bool) (*types.Header, error) {
```

Comment:
It looks like PreviousValidCoincidentOnPath is seemingly identical to RollbackToCanonical. Why is more "PC" language preferred over something that is more explicit and takes an action?

Reply:
It feels as if the CheckPCCRC and PCCOP logic is extraneous due to the ability to receive a termini regardless of canonicalization. Due to the decoupling I feel as if these two should go away and work as follows:

PCRC: 
- Run all the subordinate checks using PCOP in the context in which they are found. 
- No need to check canonical as that has been delegated to the subordinates to find outside of PCRC.
- All termini are returned and errors are returned based on the returned PCOP values.

PCOP:
- Subordinates obtain the latest coincident block to them at a given order to meet the dominant PCRC call.
- This coincident check should return a terminus regardless of it is canonical or not **due to the decoupled canonical checks**.

Reply:
IMO this simplifies the logic and will make it easier to understand.